src/imitation/policies/replay_buffer_wrapper.py

```diff
@@ -0,0 +1,83 @@
+"""Wrapper for reward labeling for transitions sampled from a replay buffer."""
+
+
+from typing import Mapping, Optional, Type
+
+import numpy as np
+import torch as th
+from stable_baselines3.common.buffers import ReplayBuffer
+from stable_baselines3.common.type_aliases import ReplayBufferSamples
+
+from imitation.rewards.reward_function import RewardFn
+from imitation.util import util
+
+
+def _samples_to_reward_fn_input(
+    samples: ReplayBufferSamples,
+) -> Mapping[str, np.ndarray]:
+    """Convert a sample from a replay buffer to a numpy array."""
+    return dict(
+        state=samples.observations.cpu().numpy(),
+        action=samples.actions.cpu().numpy(),
+        next_state=samples.next_observations.cpu().numpy(),
+        done=samples.dones.cpu().numpy(),
+    )
+
+
+class ReplayBufferRewardWrapper(ReplayBuffer):
+    """Relabel the rewards in transitions sampled from a ReplayBuffer."""
+
+    def __init__(
+        self,
+        *args,
+        replay_buffer_cls: Type[ReplayBuffer] = ReplayBuffer,
+        reward_fn: Optional[RewardFn] = None,
```

Comment:
When do we want this to be `None`? Can't we just not wrap the replay buffer in that case?

Reply:
This is actually pretty annoying. Because ReplayBuffer is created by https://github.com/DLR-RM/stable-baselines3/blob/c1f1c3d3d796054f68b3fa741c1a0be4ce2187b5/stable_baselines3/common/off_policy_algorithm.py#L212

```python
self.replay_buffer = self.replay_buffer_class(
    self.buffer_size,
    self.observation_space,
    self.action_space,
    device=self.device,
    n_envs=self.n_envs,
    optimize_memory_usage=self.optimize_memory_usage,
    **self.replay_buffer_kwargs,
)
```
so we need to set reward_fn as a kwarg rather than an arg, otherwise an error of `TypeError: got multiple values for argument` will occur. That's why I set default values to `reward_fn` and `replay_buffer_class`. For now I'm not aware of a better way of working around this, happy to take ideas.

Reply:
I'm not sure I quite understand this. Keyword arguments don't need to have default values. You can force a keyword argument by including a `*`: `def f(x,*,y)` requires that `y` be specified as a kwarg.

Reply:
Sounds good! This is exactly what I'm asking about. Thanks